homeassistant/components/google_domains/__init__.py

```diff
@@ -42,35 +42,47 @@ async def async_setup(hass, config):
 
     session = hass.helpers.aiohttp_client.async_get_clientsession()
 
-    result = await _update_google_domains(
-        hass, session, domain, user, password, timeout
+    address = await _update_google_domains(
+        hass, session, domain, user, password, timeout, None
     )
 
-    if not result:
+    if address is None:
         return False
 
     async def update_domain_interval(now):
-        """Update the Google Domains entry."""
-        await _update_google_domains(hass, session, domain, user, password, timeout)
+        nonlocal address
+        address = await _update_google_domains(
+            hass, session, domain, user, password, timeout, address
+        )
 
     hass.helpers.event.async_track_time_interval(update_domain_interval, INTERVAL)
 
     return True
 
 
-async def _update_google_domains(hass, session, domain, user, password, timeout):
-    """Update Google Domains."""
-    url = f"https://{user}:{password}@domains.google.com/nic/update"
+async def _update_google_domains(
+    hass, session, domain, user, password, timeout, address
+):
+    try:
+        # Check IP address
+        with async_timeout.timeout(timeout):
+            resp = await session.get("https://api.ipify.org")
```

Comment:
The docs referenced in the PR also mention https://domains.google.com/checkip as a source for how to get your current ip.  Might it be more consistent with this component to use Google's checkip?  (I'm not familiar with ipify.org or anything, but just noticed that it seems kind of arbitrary which public service to use.  Using google's might be consistent with this component).  Does it also suffer the same problem you mentioned?

Reply:
I probably should have mentioned that. Yes, it has the same problem. I chose ipify.org because some other integrations use it (notably route53).